### PR TITLE
Feature/reference block

### PR DESCRIPTION
**Pull request by HugoLaguna**
#### What problem is this solving?

Reference product block are missing, but it too are missing on context types

#### How to test it?

I'm sending type of reference on context 
https://github.com/vtex-apps/product-summary-context/pull/20

## Workspace

Example with a block of product-summary 

- https://reference--idville.myvtex.com/

## Evidence

image

### CODE REVIEW

**Comment by HugoLaguna** [comment]
Minor (backwards-compatible functionality)

File: react/ProductSummaryReference.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+import { ProductSummaryContext } from 'vtex.product-summary-context'
+
+const { useProductSummary } = ProductSummaryContext
+
+const CSS_HANDLES = ['referenceContainer']
```
Comment (igorbrasileiro) [inline react/ProductSummaryReference.tsx:7]: This array is readonly
```suggestion
const CSS_HANDLES = ['referenceContainer'] as const
```

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
The code looks good to me, great job @HugoLaguna 👏 . Just add changelog and I approve

**Comment by HugoLaguna** [comment]
after deploy [type](https://github.com/vtex-apps/product-summary-context/pull/20) of `product-summary-context` i think that we can deploy it 
image

File: react/ProductSummaryReference.tsx (hunk 1 above)
Comment (HugoLaguna) [reply react/ProductSummaryReference.tsx:7]: fixed!

File: docs/README.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ Now, you are able to use all blocks exported by the `product-summary` app. Check
 | [`product-summary-image`](https://vtex.io/docs/components/all/vtex.product-summary/product-summary-image) | Renders the product image. | 
 | [`product-summary-name`](https://vtex.io/docs/components/all/vtex.product-summary/product-summary-name) | Renders the product name. | 
 | [`product-summary-sku-name`](https://vtex.io/docs/components/all/vtex.product-summary/product-summary-sku-name) | Renders the selected sku name. | 
+| [`product-summary-reference`](https://vtex.io/docs/components/all/vtex.product-summary/product-summary-reference) | Renders the product reference. |
```
Comment (victorhmp) [inline docs/README.md:36]:
```suggestion
| [`product-summary-reference`](https://vtex.io/docs/components/all/vtex.product-summary/product-summary-reference) | Renders the product's reference id. | 
```

File: docs/ProductSummaryReference.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   "product-summary.shelf": {
+    "children": [
+      "product-summary-image",
+      "product-summary-name",
++     "product-summary-reference",
+      "product-summary-attachment-list",
+      "product-summary-space",
+      "product-summary-column#1"
+    ]
+  },
```
Comment (victorhmp) [inline docs/ProductSummaryReference.md:23]:
```suggestion
     "children": [
       "product-summary-image",
       "product-summary-name",
+      "product-summary-reference",
       "product-summary-attachment-list",
       "product-summary-space",
       "product-summary-column#1"
     ]
  }
```

**Review by victorhmp — CHANGES_REQUESTED** [review]
Thanks for the PR!
Left a few suggestions, and please be sure to lint the `ProductSummaryReference.tsx` file :)

File: react/ProductSummaryReference.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
```
Comment (igorbrasileiro) [inline react/ProductSummaryReference.tsx:1]: @HugoLaguna can you lint this file?

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Great job @HugoLaguna 👏 !! 
I'm reviewing as request changes, just to remember to add the changelog entry and lint the component file.

File: react/ProductSummaryReference.tsx (hunk 4 above)
Comment (HugoLaguna) [reply react/ProductSummaryReference.tsx:1]: with `vtex.product-summary-context@0.9.0` already hasn't warnings, was missing types. can you check ?
added CHANGELOG!

File: docs/ProductSummaryReference.md (hunk 3 above)
Comment (HugoLaguna) [reply docs/ProductSummaryReference.md:23]: fixed

File: docs/ProductSummaryReference.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+📢 Use this project, [contribute](https://github.com/vtex-apps/product-summary) to it or open issues to help evolve it using [Store Discussion](https://github.com/vtex-apps/store-discussion).
+
+# Product Summary Reference
+
+_Product Summary Reference_ renders the selected SKU's name.
```
[the remark this answers is not in the document]
Comment (HugoLaguna) [reply docs/ProductSummaryReference.md:5]: 😬 sorry, i copy > paste, fixed!

File: react/ProductSummaryReference.tsx (hunk 4 above)
Comment (HugoLaguna) [reply react/ProductSummaryReference.tsx:1]: wait, are talking about this ? i don't know how fix it, i'm locking for a solution
image

File: docs/README.md (hunk 2 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

**Comment by HugoLaguna** [comment]
i run `npm run lint` but have many errors 😥
image

File: react/ProductSummaryReference.tsx (hunk 4 above)
Comment (HugoLaguna) [reply react/ProductSummaryReference.tsx:1]: i fixed this alerts 
image

File: react/ProductSummaryReference.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+const CSS_HANDLES = ['referenceContainer'] as const
+
+interface Props {
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
+}
+
+const ProductSummaryReference = ({ classes }: Props) => {
+  const { product } = useProductSummary()
+  const { handles } = useCssHandles(CSS_HANDLES, { classes })
+  const productReference: string = product.productReference ?? ''
```
Comment (victorhmp) [inline react/ProductSummaryReference.tsx:17]: Why would you need to declare the type here? Why not just let it be undefined?

```suggestion
  const productReference = product.productReference
```

Comment (HugoLaguna) [reply react/ProductSummaryReference.tsx]: sorry, i used this when i hasn't property in context, i forgot remove it. fixed

**Review by victorhmp — APPROVED** [review]

**Pull request closed without merging**